apollo-router-core/src/json/value.rs

```diff
@@ -0,0 +1,559 @@
+use super::{bytestring::ByteString, map::Map};
+use std::{
+    fmt::{self, Debug},
+    mem,
+};
+
+#[derive(Clone, Eq, PartialEq)]
+pub enum Value {
+    Null,
+    Bool(bool),
+    Number(serde_json::Number),
+    String(ByteString),
+    Array(Vec<Value>),
+    Object(Map<String, Value>),
+}
```

Comment:
:yarn: This feature makes a lot of sense imo. I know at least one project that would benefit from it. I'd suggest you extract it and make it an OSS crate under MIT OR Apache-2.0

It's also possible that someone already made this... did you check already?

Reply:
Yes, I was considering making it a separate crate. Which project are you referring to?
Surprisingly, I have not found any crate doing that already. There's [serde_bytes](https://github.com/serde-rs/bytes), but it allocates new `Bytes` instances instead of using the input one.

Reply:
My partner is working on Rust and faster version of rethinkdb-python, there is a script for importing data from json file.

Reply:
I have no update on that front yet. If you can add abritrary_precision it would be nice though (but not required)

Reply:
that should be easy to add

Reply:
famous last word lol

Reply:
it is in 0.1.3: https://crates.io/crates/serde_json_bytes
other features are deactivated for now, I'll propose some PRs to serde_json first to make its code more reusable

Reply:
Thanks I will keep you update

Reply:
He is leaving for a week so I won't have any update for a while